### PR TITLE
Update CUDA base images in CI

### PR DESCRIPTION
**Pull request by dalg24**

### CODE REVIEW

**Comment by dalg24** [comment]
It looks like the key had been updated on ubuntu20.04 based images but not ubuntu18.04 so I gave up and installed the new key manually

**Comment by aprokop** [comment]
Why Clang 13 instead of Clang 14?

**Comment by dalg24** [comment]
test this please

**Comment by masterleinad** [comment]
Retest this please.

**Comment by masterleinad** [comment]
This should finally work.

File: .jenkins/continuous.groovy [hunk 1]
```diff
@@ -176,7 +176,7 @@ pipeline {
                                     -D CMAKE_CXX_COMPILER_LAUNCHER=ccache \
                                     -D CMAKE_CXX_COMPILER=clang++ \
                                     -D CMAKE_CXX_EXTENSIONS=OFF \
-                                    -D CMAKE_CXX_FLAGS="-Wpedantic -Wall -Wextra" \
+                                    -D CMAKE_CXX_FLAGS="-Wpedantic -Wall -Wextra -DCUB_USE_COOPERATIVE_GROUPS" \
```
Comment (aprokop) [inline .jenkins/continuous.groovy:179]: Why this change?

**Review by aprokop — COMMENTED** [review]
I think there is a lot of knowledge in this PR that is not documented. It would be nice to document the process of updating the keys if they get rotated again.

File: .jenkins/continuous.groovy (hunk 1 above)
Comment (masterleinad) [reply .jenkins/continuous.groovy:179]: The full commit message for 034ec05b32deb6d089a5d289359dc1b47da385a3 is
>Workaround problematic __syncs in Thrust.
>
> See https://bugs.llvm.org/show_bug.cgi?id=43505

Without this, `Thrust` is using `shlf` without sync which was removed from PTX ISA version 6.4. According to the same bug report, this should be fixed in `Cuda 11.3` but I preferred the workaround over figuring out how to get a recent enough clang for that.

Comment (masterleinad) [reply .jenkins/continuous.groovy]: Technically, this preprocessor variable is not documented and I'm not quite sure we should always mess with it but sure, I'll give it a try and add it.

Comment (aprokop) [reply .jenkins/continuous.groovy]: https://github.com/NVIDIA/cub/pull/164

Comment (aprokop) [reply .jenkins/continuous.groovy]: > We need to do that workaround in our headers

Agreed.

File: src/details/ArborX_DetailsSortUtils.hpp [hunk 2]
```diff
@@ -25,6 +25,15 @@
 // clang-format off
 #if defined(KOKKOS_ENABLE_CUDA)
 #  if defined(KOKKOS_COMPILER_CLANG)
+
+// Some Thrust (or CUB to be more precise) versions use __shfl instead of
+// __shfl_sync which was removed in PTX ISA version 6.4, also see
+// https://bugs.llvm.org/show_bug.cgi?id=43505.
+// For some reason, the issue could only be observed with clang
```
Comment (aprokop) [inline src/details/ArborX_DetailsSortUtils.hpp:32]: The issue is not observed with nvcc because the automatic enabling of `CUB_USE_COOPERATIVE_GROUPS` is guarded with `__CUDACC_VER_MAJOR__` check that is only defined in nvcc.
See [this](https://github.com/NVIDIA/cub/pull/164#issuecomment-495502027) comment.

Comment (masterleinad) [reply src/details/ArborX_DetailsSortUtils.hpp:32]: I decided to check the cub version instead.

**Review by aprokop — APPROVED** [review]
LGTM

**Comment by dalg24** [comment]
@masterleinad do you mind cleaning up the history?

**Comment by masterleinad** [comment]
> @masterleinad do you mind cleaning up the history?

Sure.

**Comment by masterleinad** [comment]
Cleaned.

**Comment by dalg24** [comment]
Thank you

**Comment by dalg24** [comment]
Failure is due to warnings from building the SYCL image.  Merge as is?

**Comment by aprokop** [comment]
> Failure is due to warnings from building the SYCL image. Merge as is?

Aren't there a lot of warnings coming from building Kokkos. Still technically from building the image, so I would be OK, but shouldn't those be fixed? Or would they be fixed with the Kokkos update to 3.5?

**Comment by dalg24** [comment]
They are not our problem (we = ArborX)
